src/gui/EditWidgetIcons.cpp

```diff
@@ -141,30 +141,42 @@ void EditWidgetIcons::setUrl(const QString &url)
     abortFaviconDownload();
 }
 
-void EditWidgetIcons::downloadFavicon()
+void EditWidgetIcons::downloadFavicon(QUrl url)
```

Comment:
After reviewing the flow of the code I do not like how this function is used in very different contexts. For example, on first use (no url provided) it kicks off the favicon and redirection code block. One subsequent uses it either follows a redirect or uses the Google URL provided as a last resort. The main purpose of this function should be to kickstart the download process. Different functions (private) should be defined to handle the different behavior. Or just bake into the onRequestFinished call since it is just a single line of code.